src/bin/cli/mod.rs

```diff
@@ -682,3 +697,41 @@ fn explicit_args(matches: &ArgMatches) -> Vec<&Id> {
         })
         .collect()
 }
+
+/// A helper function to check if the target name provided by the user matches the target name in solang.toml
+/// Returns false if the target names do not match
+/// Returns true if the target names match or if solang.toml does not exist in the current directory
+fn check_target_match(target_name: &str) -> bool {
+    // Get the manifest path from the current directory
+    let manifest_path = PathBuf::from("solang.toml");
+
+    // Check if the manifest file exists
+    // If it doesn't, then we don't need to check the target name
+    if !manifest_path.exists() {
+        return true;
+    }
+
+    // Read the contents of the solang.toml file
+    let mut file = File::open(&manifest_path).expect("Failed to open the manifest");
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)
+        .expect("Failed to read solang.toml");
+
+    // Parse the TOML contents and extract the target name
+    let parsed_toml: toml::Value = toml::from_str(&contents).expect("Failed to parse solang.toml");
+    let config_target = parsed_toml["target"]["name"]
+        .as_str()
+        .expect("target name is not a string")
```

Comment:
3 unnecessary panics in 3 lines of code; please change this function to return an error instead

Reply:
Yes, they are all [already defined](https://github.com/hyperledger/solang/blob/main/src/bin/cli/mod.rs#102)